### DISCUSSION TITLE
What is the max size of attachments?

### DISCUSSION THREAD

**Discussion by Candyxxy** [CorentinTh/enclosed#126] [Q&A] [answered]
Hello, im interested in selfhosting this. 

- What is the max size of attachments?
I know we can add multiple attachments, but im curious if there is a file size limit or if we can adjust that limit if it exists.

Thanks :)

**Comment by CorentinTh** [comment, marked as answer]
Hey @Candyxxy  :wave: 
By default the size limit is set to 50MiB for the **whole encrypted note**, including the attachments and the text content. The limit is enforced by the server, and can be configured through the `NOTES_MAX_ENCRYPTED_PAYLOAD_LENGTH` environment variable.

Note that for really huge file, the encryption process on client can take a bite of time, a simple loading mechanism will be added.

You can see the details for the configuration here : https://docs.enclosed.cc/self-hosting/configuration

**Reply by benjaminjunior** [reply]
hey @CorentinTh 
thank you for putting this together. I increased the NOTES_MAX_ENCRYPTED_PAYLOAD_LENGTH to 5Gigs, but I'm still getting an error in a red boz saying: "An error occurred while creating the note, please try again.". 

here's my docker-compose file:
`services:
  enclosed:
    image: corentinth/enclosed
    ports:
      - 8787:8787
    volumes:
      - enclosed-data:/app/.data
    environment:
      - SERVER_API_ROUTES_TIMEOUT_MS=240000
      - NOTES_MAX_ENCRYPTED_PAYLOAD_LENGTH=5368709120
      - AUTHENTICATION_JWT_SECRET="*random string"
    restart: unless-stopped

volumes:
  enclosed-data:
    driver: local`

Could you please assist?
